### PR TITLE
Fix KIND_GPU

### PR DESCRIPTION
**Pull request by Tabrizian**
Closes triton-inference-server/server#2157

### CODE REVIEW

File: src/python.cc [hunk 1]
```diff
@@ -57,7 +49,9 @@
 #include "triton/core/tritonbackend.h"
 #include "triton/core/tritonserver.h"
 
-namespace triton { namespace backend { namespace python {
+namespace triton {
```
Comment (CoderHam) [inline src/python.cc:52]: Can you re-run the formatter? These three should be a single line.

File: src/python.cc [hunk 2]
```diff
@@ -393,16 +387,22 @@ ModelInstanceState::~ModelInstanceState() {
...
-  unlink(domain_socket_.substr(7).c_str());
 
-  // FIXME currently GRPC client uses an async thread for cleaning up and
-  // shutting down the connection, however, as reported in
-  // https://github.com/grpc/grpc/issues/22479 the clean up thread may continue
-  // to live after resources have been deallocated an cause a segfault. This is
-  // a workaround to do a blocking shutdown of the GRPC client
-  grpc_shutdown_blocking();
+  // Check if the domain socket has been set.
+  if (domain_socket_.length() >= 7) {
```
Comment (CoderHam) [inline src/python.cc:392]: The domain socker it is either never set i.e. "" (if the temp dir cannot be created) or set to `unix://tmp_dir_name/unix.socket` Shouldn't this simply check if it is empty?

**Review by CoderHam — CHANGES_REQUESTED** [review]

File: src/python.cc (hunk 1 above)
Comment (Tabrizian) [reply src/python.cc:52]: I rerun the formatter multiple times. Not sure why it keeps changing these into multiple lines. I can manually put them together.

File: src/python.cc (hunk 2 above)
Comment (Tabrizian) [reply src/python.cc:392]: Sure, I will change it to a simple empty check.

File: src/python.cc [hunk 3]
```diff
@@ -393,16 +385,22 @@ ModelInstanceState::~ModelInstanceState() {
...
-  // FIXME currently GRPC client uses an async thread for cleaning up and
-  // shutting down the connection, however, as reported in
-  // https://github.com/grpc/grpc/issues/22479 the clean up thread may continue
-  // to live after resources have been deallocated an cause a segfault. This is
-  // a workaround to do a blocking shutdown of the GRPC client
-  grpc_shutdown_blocking();
+  // Check if the domain socket has been set.
+  if (domain_socket_. != "") {
+    // We want to remove "unix://" from the beginning of domain_socket_
+    unlink(domain_socket_.substr(7).c_str());
```
Comment (deadeyegoodwin) [inline src/python.cc:392]: Replace "magic" 7 with strlen("unix://")

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

File: src/python.cc (hunk 1 above)
Comment (CoderHam) [reply src/python.cc]: That's odd. Let it be as the formatter changes it. This compiles successfully? Such a formatting usually indicates ill formed "{}" pairs

Comment (Tabrizian) [reply src/python.cc]: It has been compiling for 2 months.

File: src/python.cc [hunk 4]
```diff
@@ -393,16 +385,22 @@ ModelInstanceState::~ModelInstanceState() {
...
-  unlink(domain_socket_.substr(7).c_str());
 
-  // FIXME currently GRPC client uses an async thread for cleaning up and
-  // shutting down the connection, however, as reported in
-  // https://github.com/grpc/grpc/issues/22479 the clean up thread may continue
-  // to live after resources have been deallocated an cause a segfault. This is
-  // a workaround to do a blocking shutdown of the GRPC client
-  grpc_shutdown_blocking();
+  // Check if the domain socket has been set.
+  if (domain_socket_. != "") {
```
[the remark this answers is not in the document]
Comment (Tabrizian) [reply src/python.cc:390]: Sorry about this, I forgot to add my changes to the new commit.

**Comment by CoderHam** [comment]
LGTM. I think you need to re-run the CI since L0_backend_python failed the last time.

**Review by deadeyegoodwin — APPROVED** [review]

File: src/python.cc (hunk 1 above)
Comment (tanmayv25) [reply src/python.cc:53]: I believe you need .clang-format configuration file in the repo for proper formatting. Look here: https://github.com/triton-inference-server/server/blob/master/.clang-format

Comment (Tabrizian) [reply src/python.cc:53]: @tanmayv25 Thanks for pointing this out. After using the .clang-format this issue was resolved!

**Pull request merged**